src/stan/services/sample/hmc_nuts_diag_e_adapt.hpp

```diff
@@ -19,6 +19,26 @@ namespace stan {
 namespace services {
 namespace sample {
 
+namespace internal {
+/**
+ * Specializtion to get a const reference to the underlying value in a
+ *  shared_ptr.
+ */
+template <typename Context>
+inline const auto& get_context(const std::shared_ptr<Context>& x) {
```

Comment:
No so this is a weird thing I'm sorting out about cmdstan. Since cmdstan actually wraps the `var_context` for inits inside of a `shared_ptr<>` we need to access the underlying pointer. Because in cmdstan we are going to pass a `std::vector<shared_ptr<var_context>>` and the current version just passes the underlying of the shared pointer. Though I might be able to get rid of this so I'll check